lib/plugin/plugin.cpp

```diff
@@ -250,7 +250,7 @@ std::vector<std::filesystem::path> Plugin::getDefaultPluginPaths() noexcept {
   Dl_info DLInfo;
   int Status =
       dladdr(reinterpret_cast<void *>(Plugin::getDefaultPluginPaths), &DLInfo);
-  if (Status != 0) {
+  if (Status != 0 && DLInfo.dli_fname != nullptr) {
```

Comment:
We should have this condition inside the `Status != 0`.
You can find that the `else` only handles the `DLInfo.dli_fname == nullptr` with the given error message. However, one possible failure is that the `dladdr` fails and returns 0.